Cython/Build/Dependencies.py

```diff
@@ -721,6 +721,25 @@ def transitive_merge_helper(self, node, extract, merge, seen, stack, outgoing):
         finally:
             del stack[node]
 
+    def write_depfile_for(self, source, c_file):
+        dependencies = self.all_dependencies(source)
+        src_base_dir, _ = os.path.split(source)
+        cwd = os.getcwd()
+        if not src_base_dir.endswith(os.sep):
+            src_base_dir += os.sep
+        # paths below the base_dir are relative, otherwise absolute
+        paths = []
+        for fname in dependencies:
+            if fname.startswith(src_base_dir):
+                paths.append(os.path.relpath(fname, cwd))
+            else:
+                paths.append(os.path.abspath(fname))
+
+        depline = os.path.relpath(c_file, cwd) + ": \\\n  "
+        depline += " \\\n  ".join(paths) + "\n"
+
+        with open(c_file+'.dep', 'w') as outfile:
+            outfile.write(depline)
```

Comment:
Excluding the call to `self.all_dependencies()`, I think this can be moved to `Utils.py` where most of the file related utilities live. It's generally a good idea to separate calculation (i.e. figuring out the dependencies) from representation (i.e. writing them down into a dependency file).